astroquery/dastcom5/core.py

```diff
@@ -0,0 +1,340 @@
+from __future__ import print_function
+import re
+import os
+import urllib.request
+import zipfile
+import numpy as np
+import pandas as pd
+from tqdm import tqdm
+
+from astropy.time import Time
+from astropy.table import Table
+import astropy.units as u
+
+from . import conf
+from ..query import BaseQuery
+from ..utils import async_to_sync
+
+
+@async_to_sync
+class Dastcom5Class(BaseQuery):
+    """
+    Class for querying NEOs orbit from DASTCOM5.
+    """
+
+    def __init__(self, spk_id=None, name=None):
+        """
+        Parameters:
+        ----------
+        api_key : str
+            NASA OPEN APIs key (default: `DEMO_KEY`)
+        """
+        super(Dastcom5Class, self).__init__()
+        self.spk_id = spk_id
+        self.name = name
+        self.local_path = conf.ASTROQUERY_LOCAL_PATH
+        self.dbs_path = conf.DBS_LOCAL_PATH
+        self.ast_path = conf.AST_DB_PATH
+        self.com_path = conf.COM_DB_PATH
+        self.com_dtype = conf.COM_DTYPE
+        self.ast_dtype = conf.AST_DTYPE
+        self.ftp_url = conf.FTP_DB_URL
+        self._show_download_progress = self._Show_Download_Progress
+
+    def download_dastcom5(self):
+        """Downloads DASTCOM5 database.
+
+        Downloads and unzip DASTCOM5 file in default astroquery path (~/.astroquery).
+
+        """
+
+        dastcom5_dir = os.path.join(self.local_path, "dastcom5")
+        dastcom5_zip_path = os.path.join(self.local_path, "dastcom5.zip")
+
+        if os.path.isdir(dastcom5_dir):
+            raise FileExistsError(
+                "dastcom5 is already created in " +
+                os.path.abspath(dastcom5_dir)
+            )
+        if not zipfile.is_zipfile(dastcom5_zip_path):
+            if not os.path.isdir(self.local_path):
+                os.makedirs(self.local_path)
+
+            print("Downloading datscom5.zip")
+            with self._show_download_progress(unit='B', unit_scale=True, miniters=1, desc="dastcom5.zip") as t:
+                urllib.request.urlretrieve(
```

Comment:
This is not the correct way to retrieve files in astroquery.
(1) We do not use `urllib`, we use `requests`
(2) We already have sophisticated downloading tools:
https://github.com/astropy/astroquery/blob/master/astroquery/query.py#L245
Use the `_download_file` method if you want to download a file.

Reply:
Done!

Reply:
I am facing errors like this:

```
>>> dastcom5.Dastcom5().download_dastcom5()
Downloading datscom5.zip
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/shreyas/Local Forks/astroquery/astroquery/dastcom5/core.py", line 62, in download_dastcom5
    self._download_file(url=ftp_path, local_filepath=dastcom5_zip_path)
  File "/home/shreyas/Local Forks/astroquery/astroquery/query.py", line 257, in _download_file
    auth=auth, **kwargs)
  File "/home/shreyas/Softwares/anaconda3/lib/python3.6/site-packages/requests/sessions.py", line 533, in request
    resp = self.send(prep, **send_kwargs)
  File "/home/shreyas/Softwares/anaconda3/lib/python3.6/site-packages/requests/sessions.py", line 640, in send
    adapter = self.get_adapter(url=request.url)
  File "/home/shreyas/Softwares/anaconda3/lib/python3.6/site-packages/requests/sessions.py", line 731, in get_adapter
    raise InvalidSchema("No connection adapters were found for '%s'" % url)
requests.exceptions.InvalidSchema: No connection adapters were found for 'ftp://ssd.jpl.nasa.gov/pub/ssd/dastcom5.zip'


```

Reply:
oooh... interesting... I think we have dealt with this before.... hm.  `ftp` is the issue.

Reply:
What do you suggest then? Any previous issue from which I can infer about what's happening?

Reply:
I don't suppose there's an alternative http site that hosts this?

If not, then we might need to either use the hackey `requests-ftp` (which I think we should avoid...) or revert to urllib =(

Reply:
> I don't suppose there's an alternative http site that hosts this?

No, there isn't as far as I know. @Juanlu001  Does it exists? We have always used ftp.

> If not, then we might need to either use the hackey `requests-ftp` (which I think we should avoid...) or revert to urllib =(

Revert to urllib then?

Reply:
hrm.  Let's not do that just yet, because that is a high cost to pay; we lose all of the astroquery tools we've build for caching, etc.  Let's ping the astropy slack to see if they have any ideas.

Reply:
As far as I understand, there is no HTTPS access to this file, no. Too bad...

Reply:
Okay! Am I going to ping them or you?

Reply:
Alright, the best suggestion I got was to fall back on urllib.  But, instead of doing that here, let's do it in `query.py` in `_download_file`.  i.e., in that function, have a test for `if url[:3] == 'ftp': return self._ftp_download(file)` and use `urllib` to download by FTP

Reply:
Sure, will do that!

Reply:
Done!